src/dune/lib.ml

```diff
@@ -1542,7 +1564,16 @@ module Compile = struct
       t.requires
     in
     let lib_deps_info =
-      let pps = Lib_info.pps t.info in
+      let pps =
+        (* CHECK *)
+        let db = Option.value_exn db in
+        let resolve = resolve db in
+        Preprocess.Per_module.pps
+          (Preprocess.Per_module.with_instrumentation
+             (Lib_info.preprocess t.info)
+             ~instrumentation_backend:
+               (instrumentation_backend db.instrument_with resolve))
+      in
```

Comment:
Why don't we use `t.pps` and just convert the lib values to names?

Reply:
Indeed! I had missed that. I amended as suggested, it seems to work (I also added a test exercising the library case specifically).